obspy/clients/fdsn/mass_downloader/download_helpers.py

```diff
@@ -349,19 +349,29 @@ def sanitize_downloads(self, logger):
         of the download helpers does not allow for a StationXML file with no
         data.
         """
+        from obspy.io.mseed.util import get_start_and_end_time
         # All or nothing for each channel.
         for id in self.miss_station_information.keys():
-            logger.warning("No station information could be downloaded for "
-                           "%s.%s.%s.%s. All downloaded MiniSEED files "
+            logger.warning("Station information could not be downloaded for "
+                           "%s.%s.%s.%s. MiniSEED files outside of the "
+                           "station information period "
                            "will be deleted!" % (
                                self.network, self.station, id[0], id[1]))
             channel = [_i for _i in self.channels if
                        (_i.location, _i.channel) == id][0]
             for time_interval in channel.intervals:
+                # Check that the time_interval.start and end are correct!
+                time_interval.start, time_interval.end = \
+                    get_start_and_end_time(time_interval.filename)
                 # Only delete downloaded things!
                 if time_interval.status == STATUS.DOWNLOADED:
-                    utils.safe_delete(time_interval.filename)
-                    time_interval.status = STATUS.DOWNLOAD_REJECTED
+                    # Only delete if the station data are actually missing
+                    # for this time
+                    miss_start, miss_end = self.miss_station_information[id]
```

Comment:
I just saw that this PR actually already contains pieces of what I just proposed...